### PR TITLE
docs: Update pipeline and processor docs for 7.7

### PR DESCRIPTION
**Pull request by bmorelli25**
## Motivation/summary

* Adds documentation for the new `apm_remove_span_metadata` processor
* Cleans up the original **Parse data using ingest node pipelines** documentation, and adds definitions for the four default processors in the `apm` pipeline
* Adds an **Upgrade to APM Server version 7.7** guide. This walks through the upgrade caveat that @simitt explains in https://github.com/elastic/apm-server/pull/3446#issuecomment-605875117.
* Cleans up the RUM configuration ingest pipeline information. Instead, users are directed to the updated `configuring-ingest` documentation.

## Questions

I don't quite understand [this](https://github.com/elastic/apm-server/issues/3441#issue-576922757) statement:

> One important caveat is that users with an output other than Elasticsearch need to run setup against Elasticsearch to load the new pipeline, otherwise span documents will start taking much more space.

## Docs preview

http://apm-server_3593.docs-preview.app.elstc.co/diff

## Related issues

Closes https://github.com/elastic/apm-server/issues/3441.

### CODE REVIEW

**Comment by bmorelli25** [comment]
> but the events eventually end up in Elasticsearch (e.g. APM Server -> Logstash -> Elasticsearch)

Aha! That's the part I wasn't understanding. 🙏

File: docs/configuring-ingest.asciidoc [hunk 1]
```diff
@@ -6,26 +6,36 @@
...
+Enriches Elastic's {apm-rum-ref-v}/index.html[JavaScript RUM Agent] data by
+adding user {ref}/geoip-processor.html[Geo-IP data] to the `client.geo` field.
+
+`apm_ingest_timestamp`::
+Adds an ingest timestamp for APM events.
+
+`apm_remove_span_metadata`::
+added:[7.7,"Previously hardcoded. See <<upgrading-to-77,upgrading to 7.7>>"]
+To save storage, removes metadata fields (like `host`, `kubernetes`, and `container`)
+that are already available on the parent transaction.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/configuring-ingest.asciidoc:33]: Great point

**Comment by bmorelli25** [comment]
Asked in #apm-server as well:

Our recommended upgrade path is to restart apm server with `overwrite` set to `true`. Would another feasible upgrade path be to update the `apm` pipeline directly in ES, with the PUT pipeline API?
```
PUT _ingest/pipeline/apm
```

> One important caveat is that users with an output other than Elasticsearch need to run setup against Elasticsearch to load the new pipeline, otherwise span documents will start taking much more space.

For users doing something like this: `APM Server --> Logstash --> ES`, wouldn't running `setup` against ES require users to set up output.elasticsearch?

I can test this too.. Just wanted to see if I’m missing something obvious. Editing the pipeline directly in ES seems like an easier upgrade for non ES output users.

**Comment by axw** [comment]
@bmorelli25 yes, that would work too, it's just a bit involved and exposes the details of the processor, which could change over time.

Another option would be to run the `apm-server setup --pipelines` command, setting Elasticsearch config on the command line. e.g. if you normally use Logstash output, then you could run:

```
apm-server setup --pipelines \
  -E output.logstash.enabled=false \
  -E output.elasticsearch.enabled=true \
  -E output.elasticsearch.hosts=["http://username:password@elasticsearch:9200"]
```

**Comment by bmorelli25** [comment]
> apm-server setup --pipelines \
>   -E output.logstash.enabled=false \
>   -E output.elasticsearch.enabled=true \
>   -E output.elasticsearch.hosts=["http://username:password@elasticsearch:9200"]

Yessss. Great idea. Thank you!

**Pull request merged**